module/userProjects/helper.js

```diff
@@ -1631,8 +1632,26 @@ module.exports = class UserProjectsHelper {
                         if (updatedUserProfile && updatedUserProfile.success == true && updatedUserProfile.profileMismatchFound == true) {
                             projectCreation.data.userProfile = updatedUserProfile.data;
                         }
+                        // checking the reqbody data and userLocation type matches or not
+                        for (let key in userRoleInformation) {
+                            // Skip role validation
+                            if (key === 'role') {
```

Comment:
@ankitpws can we check userRoleInformation roles and if these roles are matching then only allow to start?

Reply:
@aks30  We can omit this step, as during our review in NVSK, we noticed that the userProfile was absent. To prevent this scenario, we included validation. Additionally, we should ensure that empty records are not present. Role checks will be conducted in the above code .